app/configurator/configurator-state.tsx

```diff
@@ -202,82 +202,147 @@ export const getFilterValue = (
+      // they need to be handled in a different way.
+      const hiddenFieldIris = getHiddenFieldIris(fields);
+      const groupedDimensionIris = getGroupedFieldIris(fields);
+
+      const isHidden = (iri: string) => hiddenFieldIris.has(iri);
+      const isGrouped = (iri: string) => groupedDimensionIris.has(iri);
+
+      dimensions.forEach((dimension) =>
+        applyTableDimensionToFilters({
+          filters,
+          dimension,
+          isHidden: isHidden(dimension.iri),
+          isGrouped: isGrouped(dimension.iri),
+        })
+      );
+    } else {
+      const fieldDimensionIris = getFieldComponentIris(fields);
+      const isField = (iri: string) => fieldDimensionIris.has(iri);
+
+      dimensions.forEach((dimension) =>
+        applyNonTableDimensionToFilters({
+          filters,
+          dimension,
+          isField: isField(dimension.iri),
+        })
+      );
+    }
 
     return chartConfig;
   }
 );
 
-export const applyDimensionToFilters = ({
+export const applyTableDimensionToFilters = ({
   filters,
   dimension,
-  isField,
   isHidden,
   isGrouped,
 }: {
   filters: Filters;
   dimension: DimensionMetaDataFragment;
-  isField: boolean;
   isHidden: boolean;
   isGrouped: boolean;
 }) => {
-  if (dimension.isKeyDimension) {
-    const f = filters[dimension.iri];
-
-    if (f !== undefined) {
-      // Fix wrong filter type
-      if (f.type === "single" && isField) {
-        delete filters[dimension.iri];
-      } else if (f.type === "multi") {
-        if (isHidden && !isGrouped) {
+  const currentFilter = filters[dimension.iri];
+  const shouldBecomeSingleFilter = isHidden && !isGrouped;
+
+  if (currentFilter) {
+    switch (currentFilter.type) {
+      case "single":
+        if (!shouldBecomeSingleFilter) {
+          delete filters[dimension.iri];
+        }
+        break;
+      case "multi":
+        if (shouldBecomeSingleFilter && dimension.isKeyDimension) {
           filters[dimension.iri] = {
             type: "single",
-            value: Object.keys(f.values)[0],
+            value:
+              Object.keys(currentFilter.values)[0] || dimension.values[0].value,
           };
         }
-      } else if (f.type === "range") {
-        if (isHidden) {
+        break;
+      case "range":
+        if (shouldBecomeSingleFilter) {
           filters[dimension.iri] = {
             type: "single",
-            value: f.from,
+            value: currentFilter.from,
           };
         }
-      }
-    } else {
-      // Add filter for this dim if it's not one of the selected multi filter fields, but
-      // only when the dimension isn't grouped, as only then we need to switch to a single filter.
-      if ((isHidden || !isField) && !isGrouped) {
-        filters[dimension.iri] = {
-          type: "single",
-          value: dimension.values[0].value,
-        };
-      }
+        break;
+      default:
+        throw new Error();
```

Comment:
Is this here to check that this case should in fact never happen? If so, I recommend using TypeScript to do the work for us instead of throwing at runtime: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#exhaustiveness-checking

Reply:
Yes - never used it previously, thanks!